editor/animation_track_editor.cpp

```diff
@@ -7759,3 +7912,1208 @@ AnimationTrackKeyEditEditor::AnimationTrackKeyEditEditor(Ref<Animation> p_animat
+	moving_selection_offset = 0;
+}
+
+void AnimationMarkerEdit::_move_selection(float p_offset) {
+	moving_selection_offset = p_offset;
+	queue_redraw();
+}
+
+void AnimationMarkerEdit::_move_selection_commit() {
+	EditorUndoRedoManager *undo_redo = EditorUndoRedoManager::get_singleton();
+	undo_redo->create_action(TTR("Animation Move Markers"));
+
+	for (HashSet<StringName>::Iterator E = selection.last(); E; --E) {
+		StringName name = *E;
+		double time = animation->get_marker_time(name);
+		float newpos = time + moving_selection_offset;
+		undo_redo->add_do_method(animation.ptr(), "remove_marker", name);
+		undo_redo->add_do_method(animation.ptr(), "add_marker", name, newpos);
+		undo_redo->add_do_method(animation.ptr(), "set_marker_color", name, animation->get_marker_color(name));
+		undo_redo->add_undo_method(animation.ptr(), "remove_marker", name);
+		undo_redo->add_undo_method(animation.ptr(), "add_marker", name, time);
+		undo_redo->add_undo_method(animation.ptr(), "set_marker_color", name, animation->get_marker_color(name));
+
+		// add_marker will overwrite the overlapped key on the redo pass, so we add it back on the undo pass.
+		if (StringName overlap = animation->get_marker_at_time(newpos)) {
+			if (select_single_attempt == overlap) {
+				select_single_attempt = "";
+			}
+			undo_redo->add_undo_method(animation.ptr(), "add_marker", overlap, newpos);
+			undo_redo->add_undo_method(animation.ptr(), "set_marker_color", overlap, animation->get_marker_color(overlap));
+		}
+	}
+
+	moving_selection = false;
+	undo_redo->add_do_method(this, "_stop_animation_player_section_preview");
+	undo_redo->add_undo_method(this, "_stop_animation_player_section_preview");
+	undo_redo->add_do_method(this, "queue_redraw");
+	undo_redo->add_undo_method(this, "queue_redraw");
+	undo_redo->commit_action();
+	_update_key_edit();
+}
+
+void AnimationMarkerEdit::_delete_selected_markers() {
+	if (selection.size()) {
+		EditorUndoRedoManager *undo_redo = EditorUndoRedoManager::get_singleton();
+		undo_redo->create_action(TTR("Animation Delete Keys"));
+		for (const StringName &name : selection) {
+			double time = animation->get_marker_time(name);
+			undo_redo->add_do_method(animation.ptr(), "remove_marker", name);
+			undo_redo->add_undo_method(animation.ptr(), "add_marker", name, time);
+			undo_redo->add_undo_method(animation.ptr(), "set_marker_color", name, animation->get_marker_color(name));
+		}
+		_clear_selection_for_anim(animation);
+
+		undo_redo->add_do_method(this, "queue_redraw");
+		undo_redo->add_undo_method(this, "queue_redraw");
+		undo_redo->commit_action();
+		_update_key_edit();
+	}
+}
+
+void AnimationMarkerEdit::_move_selection_cancel() {
+	moving_selection = false;
+	queue_redraw();
+}
+
+void AnimationMarkerEdit::_clear_selection(bool p_update) {
+	AnimationPlayer *player = AnimationPlayerEditor::get_singleton()->get_player();
+	if (player && player->is_playing()) {
+		player->pause();
+		player->reset_current_section();
+	}
```

Comment:
Certainly, `pause()` can be a stopgap to avoid breaking the state after `set_current_section()`, but if this is always called at the same time as `set_current_section()`, there is no reason not to include it in `set_current_section()`.

If `set_current_section()` breaks without `pause()`, then `set_current_section()` should not be exposed to GDScript as a core-only API. In other words, implementing `pause()` on the editor side only hides the problem, and that is where the problem exists: `set_current_section()` breaks when used at runtime. So currently, the following code does not actually set the section, which is counterintuitive:

```gdscript
$AnimationPlayer.play("new_animation")
$AnimationPlayer.set_current_section_with_markers("new_marker", "new_marker2")
```

In my opinion `set_current_section()` should be able to set the section correctly without stopping playback without `pause()`. This fix would be the last effort needed.

Reply:
After re-investigating, it seems that the settings for the section being played probably work fine at runtime, so there seems to be a problem with the section selection on the Editor side. Anyways, it needs to be fixed since there is a discrepancy in the feeling of using the editor and the runtime.